### PR TITLE
add a logger

### PR DESCRIPTION
**Pull request by ibawt**
Given the design of having no runtime impact, I thnk that error conditions will end up being eaten and not reported.  While I don't think the user of the library will need to handle the cases directly it'd be good know if for instance they screwed up the header for propagation etc etc.

@fbogsany

### CODE REVIEW

File: api/lib/opentelemetry/logger.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
+# Copyright 2019 OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+require 'logger'
+
+module OpenTelemetry
+  class << self
+    attr_accessor :logger
```
Comment (indrekj) [inline api/lib/opentelemetry/logger.rb]: would it make more sense to put it to https://github.com/open-telemetry/opentelemetry-ruby/blob/2a22a85e09a2c4823a804d71481c3e4cc606f5fd/api/lib/opentelemetry.rb which already includes similar things and it's on top level also ?

Comment (ibawt) [reply api/lib/opentelemetry/logger.rb]: we could for sure.  I started this thinking we would subclass `Logger` but I didn't see any reason at the moment. 

any opinions @fbogsany ?

Comment (fbogsany) [reply api/lib/opentelemetry/logger.rb]: Nope, no opinions.

Comment (ibawt) [reply api/lib/opentelemetry/logger.rb]: lets see what everyone thinks.  I'm ok either way.

**Comment by ibawt** [comment]
👀

File: api/lib/opentelemetry/logger.rb (hunk 1 above)
Comment (ibawt) [reply api/lib/opentelemetry/logger.rb]: I moved it in

**Pull request merged**